engine/rpcserver.go

```diff
@@ -5884,3 +5884,14 @@ func (s *RPCServer) GetCurrencyTradeURL(ctx context.Context, r *gctrpc.GetCurren
 		Url: url,
 	}, nil
 }
+
+// rpcContextToLongLivedSession converts a short-lived incoming context to a long-lived outgoing context, this is due
+// to the incoming context being cancelled when the RPC call completes.
+func (s *RPCServer) rpcContextToLongLivedSession(ctx context.Context) context.Context {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		md = metadata.New(nil) // Fallback to empty metadata
+	}
+	// TODO: Capture RPC server context for cancellations and use it here
```

Comment:
This is a very specific edgecase I'm not sure needs to be a todo, especially in the case of storing a context in a struct. Feels like trying to race against the request. 
eg
- Send SubmitOrder via client
- Server is alive, so accepts request
- Server sends SubmitOrder to exchange
- You desperately attempt to cancel the server to stop that FOMO 120x Fart/USD order
- You hope that the context cancelFunc hits before the exchange submits the order

The TODO can still stay